pkg/wallet/wallet.go

```diff
@@ -63,12 +62,58 @@ func (w *BaseWallet) DecryptMessage(encMessage []byte, toVerKey string) ([]byte,
 
 // PackMessage Pack a message for one or more recipients.
 func (w *BaseWallet) PackMessage(envelope *Envelope) ([]byte, error) {
```

Comment:
As discussed in the planning meeting, the abstraction of Pack/Unpack and underlying crypto primitives in the wallet should be removed.

More specifically, does the wallet need to know about envelopes?